processor/eventfilter/eventfilter.go

```diff
@@ -69,3 +77,325 @@ func GetSupportedMessageEvents(destination *backendconfig.DestinationT) ([]strin
+		5. if sourceType.connectionMode = {} -- rely on default behaviour
+		6. "device" is defined as a key for a sourceType -- don't send the event
+		7. if eventProperty not in sourceType.connectionMode -- rely on property based default behaviour
+		8. if sourceType.connectionMode.eventProperty = {} (both allowAll, allowedValues are not included in defConfig) -- rely on default behaviour
+		9. if sourceType.connectionMode.eventProperty = { allowAll: true, allowedValues: ["track", "page"]} (both allowAll, allowedValues are included in defConfig) -- rely on default behaviour
+	*/
+
+	destConnModeI := misc.MapLookup(destination.Config, "connectionMode")
+	if destConnModeI == nil {
+		pkgLogger.Debug("connectionMode not present, filtering event based on default behaviour")
+		return evaluatedDefaultBehaviour, nil
+	}
+	destConnectionMode, isDestConnModeString := destConnModeI.(string)
+	if !isDestConnModeString || destConnectionMode == "device" {
+		// includes Case 6
+		pkgLogger.Errorf("Provided connectionMode(%v) is in wrong format or the mode is device", destConnModeI)
+		return false, nil
+	}
+
+	eventProperties := getSupportedEventPropertiesMap(destination.DestinationDefinition.Config, srcType, destConnectionMode, evaluatedDefaultBehaviour)
+	if len(eventProperties.EventPropsMap) == 0 {
+		return eventProperties.Allow, eventProperties.FailedResponse
+	}
+
+	// Flag indicating to let the event pass through
+	allowEvent := evaluatedDefaultBehaviour
+	for eventProperty, supportedEventVals := range eventProperties.EventPropsMap {
+		eventPropMap, isMap := supportedEventVals.(map[string]interface{})
+
+		if !allowEvent || !isMap {
+			pkgLogger.Debugf("Previous evaluation of allowAll is false or type assertion failed for an event property(%v), filtering event based on default behaviour", eventProperty)
+			allowEvent = evaluatedDefaultBehaviour
+			break
+		}
+		if eventProperty == "messageType" {
+			messageTypeEventPropDef := ConvertEventPropMapToStruct[string](eventPropMap)
+			// eventPropMap == nil  -- occurs when both allowAll and allowedVals are not defined or when allowAll contains any value other than boolean
+			// eventPropMap.AllowAll -- When only allowAll is defined
+			//  len(eventPropMap.AllowedVals) == 0 -- when allowedVals is empty array(occurs when it is [] or when data-type of one of the values is not string)
+			if eventPropMap == nil || messageTypeEventPropDef.AllowAll || len(messageTypeEventPropDef.AllowedVals) == 0 {
+				pkgLogger.Debug("Problem with message type event property map, filtering event based on default behaviour")
+				allowEvent = evaluatedDefaultBehaviour
+				continue
+			}
+
+			pkgLogger.Debugf("SupportedVals: %v -- EventType from event: %v\n", messageTypeEventPropDef.AllowedVals, messageType)
+			allowEvent = lo.Contains(messageTypeEventPropDef.AllowedVals, messageType) && evaluatedDefaultBehaviour
+		}
+	}
+	return allowEvent, nil
+}
+
+type EventPropsTypes interface {
+	~string
```

Comment:
This is new for me, can you share doc for ~string

Reply:
https://stackoverflow.com/questions/72570147/interface-for-all-numbers